src/usb/uf2/ghostfat.c

```diff
@@ -63,7 +63,33 @@ struct TextFile {
 //
 //--------------------------------------------------------------------+
 
-#define NUM_FAT_BLOCKS CFG_UF2_NUM_BLOCKS
+#define BPB_SECTOR_SIZE           ( 512)
+#define BPB_SECTORS_PER_CLUSTER   (   1)
+#define BPB_RESERVED_SECTORS      (   1)
+#define BPB_NUMBER_OF_FATS        (   2)
+#define BPB_ROOT_DIR_ENTRIES      (  64)
+#define BPB_TOTAL_SECTORS         (0x10109) // Use a fixed (near max) number of sectors for the FAT file system
+#define BPB_MEDIA_DESCRIPTOR_BYTE (0xF8)
+#define FAT_ENTRY_SIZE            (2)
+#define FAT_ENTRIES_PER_SECTOR    (BPB_SECTOR_SIZE / FAT_ENTRY_SIZE)
+// NOTE: MS specification explicitly allows FAT to be larger than necessary
+#define BPB_SECTORS_PER_FAT       ( (BPB_TOTAL_SECTORS / FAT_ENTRIES_PER_SECTOR) + \
+                                   ((BPB_TOTAL_SECTORS % FAT_ENTRIES_PER_SECTOR) ? 1 : 0))
+#define DIRENTRIES_PER_SECTOR     (BPB_SECTOR_SIZE/sizeof(DirEntry))
+#define ROOT_DIR_SECTOR_COUNT     (BPB_ROOT_DIR_ENTRIES/DIRENTRIES_PER_SECTOR)
+
+#if defined(CFG_UF2_NUM_BLOCKS)
+  STATIC_ASSERT(BPB_TOTAL_SECTORS >= CFG_UF2_NUM_BLOCKS); // configuration used, and asked for a larger size GhostFAT?
```

Comment:
we can just used `CFG_UF2_NUM_BLOCKS` as BPB_TOTAL_SECTORS or vice versa (1 as alias to other). Since the value is used by usb_msc to response to SCSI Read Capacity command.

Reply:
Thank you for catching that!  I did not know it was used elsewhere.  👍